app/components/outside-click/component.js

```diff
@@ -0,0 +1,53 @@
+import Ember from 'ember';
+import layout from './template';
+
+export default Ember.Component.extend({
+  classNames: ['outside-click'],
+  layout,
+  isOutside: false,
+  excludedClasses: [],
+
+  onOutsideClick() {},
+
+  init() {
+    this._super(...arguments);
+    this.handleDown = this.handleDown.bind(this);
+    this.handleUp = this.handleUp.bind(this);
+  },
+
+  didInsertElement() {
+    this._super(...arguments);
+    document.addEventListener('mousedown', this.handleDown, true);
+    document.addEventListener('mouseup', this.handleUp, true);
+  },
+
+  willDestroyElement() {
+    this._super(...arguments);
+    document.removeEventListener('mousedown', this.handleDown, true);
+    document.removeEventListener('mouseup', this.handleUp, true);
+  },
+
+  handleDown(e) {
+    if (this.isDestroyed || this.isDestroying) {
+      return;
+    }
+
+    const isExcluded = this.get('excludedClasses').some((excludedClass) => {
+      return ` ${e.target.className} `.indexOf(` ${excludedClass} `) > -1;
```

Comment:
`excludedClasses` doesn't appear to be used for anything, so it can probably just be dropped.

If you want to keep it, this is not sufficient to check if `excludedClass` is a class that `e.target` has.  `excludedClasses=['a']` will match any class that contains an `a`, e.g. `foobar`.  Use `$(e.target).hasClass(excludedClass)`.